### PR TITLE
tests: verify systemd-sysuser.service with `ConditionNeedsUpdate=` will be triggered with new deployment

### PR DESCRIPTION
**Pull request by HuijingHei**
See https://github.com/ostreedev/ostree/issues/3069#issuecomment-1798115799

### CODE REVIEW

File: tests/kola/systemd/condition-needs-update [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+## kola:
+##   description: Verify systemd-sysuser.service with `ConditionNeedsUpdate=`
+##     will be triggered with new deployment at boot time.
```
Comment (jlebon) [inline tests/kola/systemd/condition-needs-update:5]: Can we mark this `platform-independent`?

File: tests/kola/systemd/condition-needs-update [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if getent passwd ${username}; then
+      fatal "should not get user ${username} after second boot"
+    fi
+    # create a new deployment
+    ref=$(rpm-ostree status --json | jq -r '.deployments[0].origin')
+    ostree admin deploy --karg-append=somedummykarg=1 $ref
+    /tmp/autopkgtest-reboot third-boot
+    ;;
+
+  third-boot)
```
Comment (jlebon) [inline tests/kola/systemd/condition-needs-update:51]: Should we be checking in this block that `somedummykarg` is present in `/proc/cmdline`?

File: tests/kola/systemd/condition-needs-update [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  second-boot)
+    # reboot to check user footest not created
+    ok "second boot"
+    if getent passwd ${username}; then
+      fatal "should not get user ${username} after second boot"
+    fi
+    # create a new deployment
+    ref=$(rpm-ostree status --json | jq -r '.deployments[0].origin')
+    ostree admin deploy --karg-append=somedummykarg=1 $ref
```
Comment (jlebon) [inline tests/kola/systemd/condition-needs-update]: Minor: this could be `rpm-ostree kargs --append=...`.

**Review by jlebon — DISMISSED** [review]
LGTM overall, thanks!

File: tests/kola/systemd/condition-needs-update (hunk 3 above)
Comment (HuijingHei) [reply tests/kola/systemd/condition-needs-update]: Sounds more reasonable, thanks!

File: tests/kola/systemd/condition-needs-update (hunk 2 above)
Comment (HuijingHei) [reply tests/kola/systemd/condition-needs-update:51]: Updated, thanks!

**Review by jlebon — APPROVED** [review]

**Comment by HuijingHei** [comment]
Thanks @jlebon for the review!

**Comment by travier** [comment]
Looks great. Thanks!

**Pull request merged**